File: ipman.shared/Entity/SearchCriteria/SiteAccountSearchCriteria.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ipman.shared.Entity.SearchCriteria
{
    public class SiteAccountSearchCriteria: SearchCriteria
    {
        public string Keyword { get; set; }
        public bool OtherUserSites { get; set; }
        public bool CurrentUserSites { get; set; }
        public string UserEmail { get; set; }
        public bool IncludeSiteAccountUserAccounts { get; set; }
        public List<Guid> ExcludedSiteAccounts { get; set; }
    }
}

File: ipman/ClientApp/entity/searchCriteria/SiteAccountSearchCriteria.ts
import { SearchCriteria } from "./SearchCriteria";

export default interface SiteAccountSearchCriteria extends SearchCriteria
{
    Keyword: string;
    OtherUserSites: boolean;
    CurrentUserSites: boolean;
    UserEmail: string;
    IncludeSiteAccountUserAccounts: boolean;
    ExcludedSiteAccounts?: string[];
    IncludedSiteAccounts?: string[];
}